.jenkins/pytorch/build.sh

```diff
@@ -30,13 +30,26 @@ cmake --version
 pip install -r requirements.txt || true
 
 if [[ "$BUILD_ENVIRONMENT" == *rocm* ]]; then
+  # This is necessary in order to cross compile (or else we'll have missing GPU device).
   export HCC_AMDGPU_TARGET=gfx900
+
+  # These environment variables are not set on CI when we were running as the Jenkins user.
+  # The HIP Utility scripts require these environment variables to be set in order to run without error.
   export LANG=C.UTF-8
   export LC_ALL=C.UTF-8
+  export PYTORCH_BUILT_WITH_ROCM=1
```

Comment:
This is misnamed, and put in the wrong place. The only place you're reading this environment variable is in the test script. So you should name this `PYTORCH_TEST_WITH_ROCM` and put it in the test script, with the rest of the envvars.

Reply:
@jithunnair-amd for awareness

Reply:
@ezyang The intention was to set the env variable if we're building with ROCm, so that when we run the unit tests, they know to skip certain unit tests for ROCm. Since $BUILD_ENVIRONMENT is defined in the CI runs, but not in the local runs, this method would allow one to define the PYTORCH_TEST_WITH_ROCM in a local run to replicate the same desired behavior. If you want to move the definition of the define to the test script, what should the condition for it be?

Reply:
Example in `.jenkins/pytorch/test.sh`

```
if [[ "$BUILD_ENVIRONMENT" == *asan* ]]; then
    export ASAN_OPTIONS=detect_leaks=0:symbolize=1
    export UBSAN_OPTIONS=print_stacktrace=1
    export PYTORCH_TEST_WITH_ASAN=1
    export PYTORCH_TEST_WITH_UBSAN=1
```

match for rocm build environment, and set `PYTORCH_TEST_WITH_ROCM=1`. If you are doing local dev outside of docker, you're expected to set the envvar yourself.

Reply:
https://github.com/ROCmSoftwarePlatform/pytorch/pull/70